core/chains/evm/client/pool.go

```diff
@@ -195,19 +196,23 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
+func (p *Pool) getBestNode() Node {
 	nodes := p.liveNodes()
+
+	// Sorting by LatestReceivedBlockNumber, because other nodes may still be syncing
+	// TODO: https://app.shortcut.com/chainlinklabs/story/50269/add-syncing-to-liveness-checker
+	// TODO: https://app.shortcut.com/chainlinklabs/story/35230/rpc-failover-should-compare-head-to-other-nodes-in-the-pool
+	sort.Slice(nodes, func(i, j int) bool {
+		return nodes[i].LatestReceivedBlockNumber() > nodes[j].LatestReceivedBlockNumber()
+	})
+
 	nNodes := len(nodes)
 	if nNodes == 0 {
 		p.logger.Critical("No live RPC nodes available")
 		return &erroringNode{errMsg: fmt.Sprintf("no live nodes available for chain %s", p.chainID.String())}
 	}
 
-	// NOTE: Inc returns the number after addition, so we must -1 to get the "current" counter
-	count := p.roundRobinCount.Inc() - 1
-	idx := int(count % uint32(nNodes))
-
-	return nodes[idx]
+	return nodes[0]
```

Comment:
Possible enhancement, track the last returned node in a field on `Pool`, and prefer that node when it is one of mulitiple 'best nodes'.

Reply:
Done